hkmc2/shared/src/test/mlscript/ups/JoinPatterns.mls

```diff
@@ -22,16 +22,18 @@ pattern SimpleJoin = ("abc" | "xyz") ~ ("def" | "")
 "abcxyzdef" is SimpleJoin
 //│ = false
 
-pattern Exponential = ("a" | "b") ~ ("c" | "d") ~ ("e" | "f") // ~ ("g" | "h") ~ ("i" | "j") ~ ("k" | "l") ~ ("m" | "n") ~ ("o" | "p") ~ ("q" | "r") ~ ("s" | "t") ~ ("u" | "v") ~ ("w" | "x") ~ ("y" | "z")
+:todo
```

Comment:
👀

Reply:
The code generated by this pattern will grow exponentially. It should be addressed in a future optimized string patterns PR.

Reply:
Why not say as much in a comment?

Reply:
> By the way, I did not expect this code to grow exponentially even without compilation 🤔 How come?

Because it concatenates many disjunctions. For example, `("a" | "b") ~ ("c" | "d") ~ ("e" | "f") ~ ("g" | "h") ~ ("i" | "j") ~ ("k" | "l")` has six disjunctions in a row, resulting in 2^6 = 64 branches.

The branch deduplication only works for the innermost consequents.

Reply:
Ouch.